modulemd/include/modulemd-2.0/modulemd-module-index.h

```diff
@@ -246,6 +246,40 @@ modulemd_module_index_update_from_custom (ModulemdModuleIndex *self,
                                           GError **error);
 
 
+/**
+ * modulemd_module_index_update_from_defaults_directory:
+ * @self: This #ModulemdModuleIndex object.
+ * @path: (in): The path to a directory containing defaults documents.
+ * @strict: (in): Whether the parser should return failure if it encounters an
+ * unknown mapping key or if it should ignore it.
+ * @overrides_subdir: (in) (nullable): If non-NULL, the name of a subdirectory
+ * of @path to search for overrides to the defaults.
+ *
+ * This function will open the directory at @path and iterate through it,
+ * adding any files with the suffix ".yaml" to @self. If @overrides_subdir is
+ * non-NULL, it will also open the named subdirectory of @path and include its
+ * contents, overriding any conflicts with the existing defaults.
+ *
+ * Note: If you need detailed information about what failed and why, it is
+ * better to implement the directory traversal yourself and use the
+ * modulemd_module_index_update_from_file() function, as it will return the
+ * failures information.
+ *
+ * Returns: TRUE if all ".yaml" files in the directory were imported
+ * successfully (this includes if no ".yaml" files were present). FALSE if one
+ * or more files could not be read successfully.
+ *
+ * Since: 2.8
+ */
+gboolean
+modulemd_module_index_update_from_defaults_directory (
+  ModulemdModuleIndex *self,
+  const gchar *path,
+  gboolean strict,
+  const gchar *overrides_subdir,
```

Comment:
What is the reasoning behind forcing overrides to be in a subdirectory of the provided `path`? Wouldn't it be more flexible to require the caller to provide an absolute path that could be anywhere? Or, perhaps, if the overrides began with `/` treat it as absolute, and anything else would be relative to `path`.

Reply:
I was basing it around our expected usage, but I suppose you're right: it would be more sensible to make the `overrides_subdir` be a full path. Making it relative to `path` seems like unnecessary complexity. If a relative path is provided, it should be relative to `$CWD` to avoid confusion.